### PR TITLE
Add React frontend core + local (offline) game mode

### LINKED ISSUE #7: 6. React frontend core + local (offline) game mode
Part of the Supabase-native rewrite.

Scaffold `apps/web` into a real app: Vite + React + TypeScript + Tailwind + a small component set (e.g. shadcn/ui) for a proper visual design. **Mobile-first, not just mobile-responsive** — this is a link-shared casual game, primarily opened on phones (including from chat-app in-app browsers, see issue #8's interstitial). React Router for routing.

**Visual direction:** young, playful, game-like — bold colors, rounded shapes, a bit of animation/microinteraction on moves and wins. This is an explicit break from the legacy Blazor UI's look; do not port its SVG board/color language, redesign it.

**Routing/UX is also a deliberate break from the legacy app** (which opened straight into a local game at `/`, with a separate `/lobby` mode-picker page):
- `/` is a **home menu**, not the game board. Three mode options, presented as cards/buttons:
  - **Play locally** — 2 players, same device. Navigates to `/local`.
  - **Play online with a friend** — opens a dialog (Create game / Join by code). The dialog's behavior and the resulting online game screens are issue #8's scope; this issue only needs the entry point (button + dialog shell) wired up.
  - **Play vs bot** — visible but disabled with a "coming soon" badge; not implemented yet.
- Local 2-player game mode moves to `/local` (previously `/`), using `packages/game-engine` directly client-side (no network).
- There is intentionally no `/lobby` route — the online dialog and the game room (issue #8) replace it.

Testing: component tests (Vitest + React Testing Library) for the home menu, and for the board/grid and game-info components.

Target branch: `refactor/supabase-react`.

### PR DESCRIPTION
**Pull request by zlebuh**
## Summary
- Closes #7. Scaffolds `apps/web` into a real app: Tailwind CSS v4, react-router (data mode: `createBrowserRouter`/`RouterProvider`), and a small hand-built component set (`Button`/`Card`/`Badge`/`Dialog`, the last wrapping `@radix-ui/react-dialog`) instead of running shadcn/ui's CLI, keeping the config surface minimal on a repo that runs oxlint rather than ESLint.
- New routing: `/` is a home menu with three cards — **Play locally** (→ `/local`, fully functional), **Play online with a friend** (dialog shell with disabled Create/Join placeholders — real behavior is issue #8), **Play vs bot** (disabled, "Coming soon" badge).
- `/local` is a fully playable local 2-player game, wired directly to `packages/game-engine` (first real consumer of that package from the frontend) via a new `useLocalGame` hook — no network involved. The hook works around `makeMove()`'s in-place mutation by cloning with `structuredClone` before each move, so React re-renders correctly and illegal moves (caught via `MinTacToeError`) are silently discarded.
- Local mode uses a 16×16 board matching the documented production (online) size, rather than the engine's own 20×20 default, for a consistent feel and better mobile fit (confirmed with the repo owner).
- Component tests (Vitest + RTL) for the home menu, board/grid, and game-info per CLAUDE.md's testing conventions. CI's `web` job now runs `pnpm --filter web test`.

## Notes
- Verified interactively in a real browser at a mobile viewport (375×812): placing marks, illegal-move rejection (occupied cells, mine-explosion effects), turn alternation, Restart, Home navigation, the online dialog opening/closing (including Escape), and the disabled "Play vs bot" card — not just unit tests.
- `.claude/launch.json` added so the dev server can be launched via Claude Code's preview tooling; `.claude/settings.local.json` (personal, not project config) is now gitignored.

## Test plan
- [x] `pnpm --filter web lint` — clean
- [x] `pnpm --filter web test` — 9/9 passing
- [x] `pnpm --filter web build` — clean
- [x] Manual browser verification of the golden path + edge cases at a mobile viewport (see above)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: .claude/launch.json
Comment (zlebuh) [inline .claude/launch.json:1]: Does it need to be part of the repo? I would just gitignore it.

File: apps/web/src/components/game/GameInfo.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import type { Game, Player } from '@mintactoe/game-engine'
+import { Button } from '../ui/Button'
+import { cn } from '../../lib/cn'
+
+export interface GameInfoProps {
+  game: Game
+  onReset: () => void
+}
+
+// A player is always shown as a plain color, never the O/X letter (see docs) - the letter is
```
Comment (zlebuh) [inline apps/web/src/components/game/GameInfo.tsx]: Redundnat comment

File: apps/web/src/hooks/useLocalGame.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import {
+  MinTacToeError,
+  initialize,
+  makeMove,
+  type Coordinate,
+  type Game,
+  type Rules,
+} from '@mintactoe/game-engine'
+
+// Matches the documented production (online) board size (docs/game-rules.md) rather than the
```
Comment (zlebuh) [inline apps/web/src/hooks/useLocalGame.ts]: 16x16 should be the default everywhere even in the engine

Comment (zlebuh) [reply apps/web/src/hooks/useLocalGame.ts]: (The comment is redundant)

File: packages/game-engine/scripts/parity-check.ts [hunk 3]
```diff
@@ -6,6 +6,12 @@
 // csharp-parity-fixture.json was captured by running the same scenarios below through the real
 // C# engine (a throwaway console harness referencing Zlebuh.MinTacToe.GameEngine directly).
 // Run with: pnpm --filter @mintactoe/game-engine parity-check
+//
```
Comment (zlebuh) [inline packages/game-engine/scripts/parity-check.ts]: Is this comment necessary? I dont think so

File: packages/game-engine/src/makeMove.ts [hunk 4]
```diff
@@ -70,7 +70,12 @@ export function makeMove(
   changedCoordinates.push(coordinate);
   field.player = player;
 
-  const playerWins = checkPlayerWins(game, player, coordinate);
+  // An exploded mine cell is still marked with the triggering player (matches the C# original -
+  // see GameMakeMove.cs), but it must not count as a win for them: checkPlayerWins() always
+  // credits the just-placed coordinate as "1" without re-checking whether *it* is a mine (it
+  // only checks isMine while walking outward to neighbors), so a move that only detonates a
+  // mine needs to skip the win check entirely rather than relying on that function to reject it.
+  const playerWins = !field.isMine && checkPlayerWins(game, player, coordinate);
```
Comment (zlebuh) [inline packages/game-engine/src/makeMove.ts:73]: Is the comment necessary? The !field.IsMine check is a good optimization

File: .gitignore [hunk 5]
```diff
@@ -140,3 +140,7 @@ coverage/
 .env
 .env.*
 !.env.example
+
```
Comment (zlebuh) [inline .gitignore:143]: I would add launch.json here too
